File: go-utils/RestartSystem.go
package main

import (
	"fmt"
	"os/exec"
	"time"
)

func main() {

	time.Sleep(5 * time.Second)

	cmd := exec.Command("shutdown", "-r", "now")

	output, err := cmd.CombinedOutput()
	if err != nil {
		printError(fmt.Sprintf("Failed to restart: %v. Output: %s", err, string(output)))
		return
	}

	fmt.Println("[{\"status\": \"success\", \"message\": \"Airi and system restart initiated after 5 seconds.\"}]")
}

func printError(msg string) {
	fmt.Printf("[{\"error\": \"%s\"}]\n", msg)
}

File: utils/RestartSystem.py
import os
import subprocess

from agno.tools import tool

BINARY_PATH = os.path.join(os.getcwd(), "go-utils", "RestartMode")


@tool
def restart_system() -> list:
    """
    Initiates a full system restart (reboot) after a 5-second delay.
    """
    if not os.path.exists(BINARY_PATH):
        return [
            {
                "error": f"Binary not found at {BINARY_PATH}. Please compile the Go utility."
            }
        ]

    try:
        subprocess.Popen(
            [BINARY_PATH],
            stdout=subprocess.DEVNULL,
            stderr=subprocess.DEVNULL,
            start_new_session=True,
        )

        return [
            {
                "status": "success",
                "message": "Airi and system restart initiated. System will reboot in 5 seconds.",
            }
        ]

    except Exception as e:
        return [{"error": str(e)}]
